modules/services/elasticsearch/main.tf

```diff
@@ -61,6 +61,8 @@ POLICY
 }
 
 resource aws_iam_service_linked_role "elasticsearch" {
+  # there can only be one
+  count = var.create_iam_service_linked_role ? 1 : 0
```

Comment:
great, I see you get this error because trying to create this in the same organization (and we already created them for production).

Did you notice any error when creating the `aws_elasticsearch_domain.openedx`? As that resource "depends_on" the linked_role.

I think we can safely remove the `depends_on` btw if you found potential problems there

Reply:
Is this because a naming clash between stage and prod environments? In the other setup approach, we prefixed all the resources with an environment name and put them inside separate VPC's, so there wouldn't be any naming clashes.

Reply:
@giovannicimolin this resource needs to be unique per AWS Organization, not a problem with naming

Reply:
Is this something specific to elasticsearch? In the other deployment method we're not using any service linked roles, so there might be a way around it.

We're gating access to redis and memcached through security groups, so maybe you can do the same with ES and leave it without authentication? 
Alternatively, if you need to share this resource between 2 terraform deployments, you can provision it using the same code and `terraform import` it on the newer code.

Reply:
Actually I didn't find any reference to an `aws_elasticsearch_domain` in the previous module, and in KKUx you actually installed ES inside an AWS instance which yes, doesn't need this and can only be configured with security groups.

the `aws_elasticsearch_domain` needs this policy and the security groups to work.

Reply:
@eLRuLL I think we could probably get away without this `aws_iam_service_linked_role` if we set up all the security groups ourselves, like @giovannicimolin suggested. But in the interests of time.. this was the quickest way forward.

Are you ok with this @giovannicimolin ?

Reply:
we are setting up the security groups :thinking: and the `aws_elasticsearch_domain` also has VPC access enabled, and it still needs the linked role, maybe I am missing something but I added it because it was necessary

Reply:
@eLRuLL The ES service-linked role adds a bunch of permissions to let ES manage EC2 and ELB resources for us: https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/slr-es.html

There must be something in that list of things that ES is doing for us but isn't in our resources. I'd prefer if we had everything explicitly defined here, but I'm also conscious of the setup time, so am ok to leave this as-is.